File: BackendAPI/Models/Conversation.cs
namespace BackendAPI.Models;

public class Conversation
{
    public int ConversationId { get; set; }
    public string Name { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? ColorTheme { get; set; }
    public ICollection<ConversationMember> UserList { get; set;} = new List<ConversationMember>();
}

File: frontend/src/features/conversations/types/conversation.ts
import type { User } from "../../users/types/user";

export type conversationId = number;

export interface Conversation {
  conversationId: number;
  name: string;
  createdAt: Date;
  participants: User[];
  colorTheme?: string;
}

export interface CreateConversationDto {
  name: string;
}

export interface UpdateConversationDto {
  name?: string;
}

export interface ConversationSummary {
  id: number;
  name: string | null;
}

export interface ConversationRaw {
  id: number;
  name: string;
  createdAt: string;
  members: User[];
  colorTheme?: string;
}
